core/src/main/java/org/infinispan/persistence/sifs/LogAppender.java

```diff
@@ -58,6 +59,8 @@ public class LogAppender implements Consumer<LogAppender.WriteOperation> {
    // This is only accessed by the writeProcessor thread
    private FlowableProcessor<Consumer<LogAppender>> completionProcessor;
 
+   private CompletableFuture<Void> stopped = CompletableFutures.completedNull();
```

Comment:
Should it be volatile? I guess it could be read by another thread on the complete.

Reply:
Technically it isn't needed as there is a write barrier before being processed in an rxjava Flowable and the start and stop are both synchronized, but it doesn't hurt really.

Reply:
Also in case if you are wondering the stop/start methods are not safe from multiple invocations, but that is true of almost every component and in fact many can't be restarted once stopped even.